types/styled-components/index.d.ts

```diff
@@ -59,32 +60,47 @@ export type ThemedGlobalStyledClassProps<P, T> = P & {
 export interface GlobalStyleClass<P, T>
     extends React.ComponentClass<ThemedGlobalStyledClassProps<P, T>> {}
 
-export interface StyledComponentClass<P, T, O = P>
-    extends React.ComponentClass<ThemedOuterStyledProps<O, T>> {
```

Comment:
Classes can't forward refs; this caused the type of `ref` to be impossible (even ref function or ref object were intersected with `string`)